# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: d2772a7482ed
# Full Hash: d2772a7482ed9394a2c30acc6d03caa36317e75f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-30 16:21:37
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Allow current key to be provided with PBackgroundIDBCursor::Continue. r=ttung,asuth
#   
#   The key is assumed to be unset for now.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43251
# ==============================================================================

diff -r 340dfa69aed8 -r d2772a7482ed dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Fri Sep 27 10:11:45 2019 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Fri Sep 27 10:11:41 2019 +0000
@@ -7654,8 +7654,8 @@
 
   mozilla::ipc::IPCResult RecvDeleteMe() override;
 
-  mozilla::ipc::IPCResult RecvContinue(
-      const CursorRequestParams& aParams) override;
+  mozilla::ipc::IPCResult RecvContinue(const CursorRequestParams& aParams,
+                                       const Key& key) override;
 
   bool IsLocaleAware() const { return !mLocale.IsEmpty(); }
 
@@ -15243,8 +15243,8 @@
   return IPC_OK();
 }
 
-mozilla::ipc::IPCResult Cursor::RecvContinue(
-    const CursorRequestParams& aParams) {
+mozilla::ipc::IPCResult Cursor::RecvContinue(const CursorRequestParams& aParams,
+                                             const Key& aCurrentKey) {
   AssertIsOnBackgroundThread();
   MOZ_ASSERT(aParams.type() != CursorRequestParams::T__None);
   MOZ_ASSERT(!mActorDestroyed);
@@ -15262,6 +15262,10 @@
 #endif
       ;
 
+  // TODO: For now, the child always passes an empty key. This will be changed
+  // in a subsequent patch for Bug 1168606.
+  MOZ_ASSERT(aCurrentKey.IsUnset());
+
   if (!trustParams && !VerifyRequestParams(aParams)) {
     ASSERT_UNLESS_FUZZING();
     return IPC_FAIL_NO_REASON(this);